File: ruby/word-count/word_count.rb
class Phrase
  attr_reader :word_count

  def initialize(string)
    words = stemming_word(string)
    @word_count = build_dictionary(words)
  end

  private

  def stemming_word(string)
    string.downcase.split(/[\s:;!&@$%^&,.]+/).map do |word|
      word.gsub(/^['"]|['"]$/, "")
    end
  end

  def build_dictionary(words)
    words.reduce({}) do |dic, word|
      dic.tap { |d| d[word] = d.fetch(word, 0) + 1 }
    end
  end
end

module BookKeeping
  VERSION = 1
end

File: python/word-count/word_count.py
import re

def word_count(string):
    tokens = re.split('[_\W]', string.lower())
 
    dic = {}
    for token in tokens:
        if len(token) is 0:
            continue
        dic[token] = dic.get(token, 0) + 1

    return dic
